projects/kit/components/input-number/input-number.component.ts

```diff
@@ -176,13 +191,20 @@ export class TuiInputNumberComponent
 
         this.updateValue(capped);
 
-        if (capped !== maskedNumberStringToNumber(value)) {
+        if (
+            capped !==
+            maskedNumberStringToNumber(
+                value,
+                this.numberFormatSettings.decimalSeparator,
+                this.numberFormatSettings.thousandSeparator,
+            )
+        ) {
             this.nativeValue = this.formattedValue;
         }
     }
 
     onKeyDown(event: KeyboardEvent) {
-        if (event.key !== ',' && event.key !== '.') {
+        if (!TUI_DECIMAL_SYMBOLS.includes(event.key)) {
```

Comment:
We try to stick to ES6 in our code and `Array.includes` is ES7. People might have to add separate polyfill for it, since we have to support some old Chrome versions internally.